detection_rules/rule.py

```diff
@@ -625,6 +677,14 @@ def validates_query_data(self, data, **kwargs):
         if data.get('alert_suppression') and data['type'] not in ('query', 'threshold'):
             raise ValidationError("Alert suppression is only valid for query and threshold rule types.")
 
+    def transform(self, obj: dict) -> dict:
```

Comment:
Any issue with importing here if we export from Kibana? I assume the transform method is only called if available from to_api_format if I recall? The schema's themselves should be resilient to this?

Reply:
Just to note, I think you are right @terrancedejesus, that it only gets called in `to_api_format` used in `_post_dict_conversion()` [here](https://github.com/elastic/detection-rules/blob/218c3bead6da1463b7a037f7a0ccf8d4df9b669a/detection_rules/rule.py#L997)

```python
        # rule type transforms
        self.data.transform(obj) if hasattr(self.data, 'transform') else False

```